File: Python/JSON Payload/pdfx.py
import requests
import json

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent = 2))


    uploaded_id = upload_response_json['files'][0]['id']
    pdfx_data = { "id" : uploaded_id, "output_type": "PDF/X-1a" }
    print(json.dumps(pdfx_data, indent = 2))


    print("Processing file...")
    pdfx_response = requests.post(url='https://api.pdfrest.com/pdfx',
                        data=json.dumps(pdfx_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(pdfx_response.status_code))
    if pdfx_response.ok:
        pdfx_response_json = pdfx_response.json()
        print(json.dumps(pdfx_response_json, indent = 2))

    else:
        print(pdfx_response.text)
else:
    print(upload_response.text)

File: PHP/Endpoint Examples/JSON Payload/pdfx.php
<?php
require 'vendor/autoload.php'; // Require the autoload file to load Guzzle HTTP client.

use GuzzleHttp\Client; // Import the Guzzle HTTP client namespace.
use GuzzleHttp\Psr7\Request; // Import the PSR-7 Request class.
use GuzzleHttp\Psr7\Utils; // Import the PSR-7 Utils class for working with streams.

$upload_client = new Client(['http_errors' => false]);
$upload_headers = [
  'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'content-filename' => 'filename.pdf',
  'Content-Type' => 'application/octet-stream'
];
$upload_body = file_get_contents('/path/to/file');
$upload_request = new Request('POST', 'https://api.pdfrest.com/upload', $upload_headers, $upload_body);
$upload_res = $upload_client->sendAsync($upload_request)->wait();
echo $upload_res->getBody() . PHP_EOL;

$upload_response_json = json_decode($upload_res->getBody());

$uploaded_id = $upload_response_json->{'files'}[0]->{'id'};

echo "Successfully uploaded with an id of: " . $uploaded_id . PHP_EOL;

$pdfx_client = new Client(['http_errors' => false]);
$pdfx_headers = [
  'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'Content-Type' => 'application/json'
];
$pdfx_body = '{"id":"'.$uploaded_id.'", "output_type":"PDF/X-1a"}';
$pdfx_request = new Request('POST', 'https://api.pdfrest.com/pdfx', $pdfx_headers, $pdfx_body);
$pdfx_res = $pdfx_client->sendAsync($pdfx_request)->wait();
echo $pdfx_res->getBody() . PHP_EOL;
